Core/include/Acts/Utilities/Intersection.hpp

```diff
@@ -39,144 +39,210 @@ inline std::ostream& operator<<(std::ostream& os, IntersectionStatus status) {
   ///
-  /// @param sInter is the intersection
-  /// @param sObject is the object to be instersected
+  /// @param intersections are the intersections
+  /// @param object is the object to be instersected
   template <typename T = representation_t,
             std::enable_if_t<std::is_same<T, object_t>::value, int> = 0>
-  ObjectIntersection(const Intersection3D& sInter, const object_t* sObject)
-      : intersection(sInter), object(sObject), representation(sObject) {}
+  ObjectMultiIntersection(const MultiIntersection3D& intersections,
+                          const object_t* object)
+      : m_intersections(intersections),
+        m_object(object),
+        m_representation(object) {}
 
   /// Object intersection
   ///
-  /// @param sInter is the intersection
-  /// @param sObject is the object to be instersected
-  /// @param sRepresentation is the object representation
-  ObjectIntersection(const Intersection3D& sInter, const object_t* sObject,
-                     const representation_t* sRepresentation)
-      : intersection(sInter),
-        object(sObject),
-        representation(sRepresentation) {}
-
-  /// Bool() operator for validity checking
-  explicit operator bool() const { return bool(intersection); }
-
-  /// @brief Smaller operator for ordering & sorting
-  ///
-  /// This operator will ignore the alternative, but simply
-  /// order the representing intersection
-  ///
-  /// @param oi is the source intersection for comparison
-  bool operator<(
-      const ObjectIntersection<object_t, representation_t>& oi) const {
-    return (intersection < oi.intersection);
+  /// @param intersections are the intersections
+  /// @param object is the object to be instersected
+  /// @param representation is the object representation
+  ObjectMultiIntersection(const MultiIntersection3D& intersections,
+                          const object_t* object,
+                          const representation_t* representation)
+      : m_intersections(intersections),
+        m_object(object),
+        m_representation(representation) {}
+
+  std::size_t size() const { return m_intersections.size(); }
+
+  const object_t* object() const { return m_object; }
+
+  const representation_t* representation() const { return m_representation; }
+
+  template <std::uint8_t index>
+  ObjectIntersection<object_t, representation_t> get() const {
+    return {m_intersections[index], m_object, m_representation, index};
   }
 
-  /// @brief Greater operator for ordering & sorting
-  ///
-  /// This operator will ignore the alternative, but simply
-  /// order the representing intersection
-  ///
-  /// @param oi is the source intersection for comparison
-  bool operator>(
-      const ObjectIntersection<object_t, representation_t>& oi) const {
-    return (intersection > oi.intersection);
+  ObjectIntersection<object_t, representation_t> get(std::uint8_t index) const {
+    return {m_intersections[index], m_object, m_representation, index};
```

Comment:
API with `get` both with compile-time and runtime index seems complicated. Do we need this?

Reply:
Agree, would probably use `operator[]` or `at()` for runtime as in stdlib.